edgedb/retry.py

```diff
@@ -5,21 +5,50 @@
 from . import transaction as _transaction
 
 
+class _RetryErrorInTxBegin(BaseException):
+    # In case when it failed to start() a transaction with a retryable error,
+    # the following query will raise this error so that the transaction can be
+    # retried. This is intentionally a BaseException because we don't want user
+    # code to capture this error and end up with a broken transaction.
+
+    def __init__(self, underlying):
+        self.underlying = underlying
```

Comment:
```suggestion
    def __init__(self, wrapped: BaseExceptions):
        self.wrapped = wrapped
```

Reply:
Oh the wrapped exception is a retryable `EdgeDBError` - but I'll rename, thanks!

Reply:
Yeah, add some docs too.